src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
not really. now there's no catch-all else block anymore. in particular i don't see what you gain with this. what did the error message look like when `$mc->{fail}` is undefined?